streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -1482,38 +1530,42 @@ public void shouldMaybeCommitActiveTasks() {
         final StateMachineTask task02 = new StateMachineTask(taskId02, taskId02Partitions, true);
         final Map<TopicPartition, OffsetAndMetadata> offsets2 = singletonMap(t1p2, new OffsetAndMetadata(2L, null));
         task02.setCommittableOffsetsAndMetadata(offsets2);
-        final StateMachineTask task03 = new StateMachineTask(taskId03, taskId03Partitions, false);
-        final Map<TopicPartition, OffsetAndMetadata> offsets3 = singletonMap(t1p3, new OffsetAndMetadata(3L, null));
-        task03.setCommittableOffsetsAndMetadata(offsets3);
+        final StateMachineTask task03 = new StateMachineTask(taskId03, taskId03Partitions, true);
```

Comment:
What does task03 and task04 work for?

Reply:
We have `commitNeeded` and `commitRequested` flags. To test all four possible combination how those flags can be set, we need 4 tasks, ie, we close missing test coverage adding new tasks for those cases.